### DISCUSSION TITLE
How do I allow a row to be updated by specific user?

### DISCUSSION THREAD

**Discussion by CovertCode** [pocketbase/pocketbase#4212] [Q&A] [answered]
Let’s say a user got authenticated by email + password.

Now there’s a collection `vendor` which has a field `by_user` and has relation to that user record.

Which means this vendor was created by user by_user.id

Now I want this record to be updated by only this user and admin,manager and mod user roles. There’s a field `user_level` which has admin,vendor,manager,employee

Is it possible to do it?

**Comment by ganigeorgiev** [comment, marked as answer]
Why are you comparing the "id" with the "email"? Are you using custom ids?

In any case, the question is too vague as there is not enough details about your collections structure and fields but in general for applying conditions for the current authenticated user the `@request.auth.*` field rule should  be used (see also [Filters syntax](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax)).

Assuming that `user_level` is part of the users auth collection, one example update API rule could be:
```js
@request.auth.id != "" &&
(
  @request.auth.id = by_user ||
  // either list each allowed level, eg.
  // (@request.auth.user_level = "admin" || @request.auth.user_level = "vendor" || ...)
  // or create a exclude rule for the disallowed level(s)
  @request.auth.user_level != "employee"
) &&
// (optional) if you want to disallow by_user change
// (aka. disallow the user to change the ownership and to assign it to someone else)
(@request.data.by_user:isset = false || @request.data.by_user = by_user)
```

**Reply by CovertCode** [reply]
I used only `@request.auth.email = by_user.email || (@request.auth.user_level:each = "admin" || @request.auth.user_level = "employee")` and it worked as expected.
I am expressing this rule as:
email of request user must be same as email present in field which is about to get updated OR user who made request must be admin or employee.

Do you find any drawbacks in this?

**Reply by ganigeorgiev** [reply]
I'm confused. Why do you use the "email" in the first place and not the "id" (note that the email could be optional, while the user id always exists)? Why do you also use `user_level:each` and not just `user_level` if it is not a multiple field?

**Reply by CovertCode** [reply]
You're right. Using `id` would be much better. Earlier I used `:each` because I was having multiple levels in each user.
I guess it would be better to have single level per user.
`@request.auth.id = by_user.id || (@request.auth.user_level = "admin" || @request.auth.user_level = "employee")`
This is my new API rule.

Thanks for your help.
